### PR TITLE
Abiv2 fuzzer: Make integer value generation consistent with range for…

### PR DESCRIPTION
**Pull request by bshastry**
… that integer type

In preparation for abiv2 isabelle encoder that only accepts integers in the range for that integer type.

### CODE REVIEW

**Review by ekpyron — APPROVED** [review]
To be honest, I'm really not a fan of such preprocessor magic :-). But looking at it a few times, it should do what it's supposed to and in order not to stall this, I'm fine with merging this like that - but I might at some point get back to it and suggest some template-based replacement of this ;-).

**Comment by ekpyron** [comment]
If I read the code correctly, we still won't end up with the full range of values for each type - but with a good portion of it, so it's not a big deal.

**Comment by bshastry** [comment]
> If I read the code correctly, we still won't end up with the full range of values for each type - but with a good portion of it, so it's not a big deal.

I agree. I think the main source of non uniform randomness is the use of a monotonically increasing counter. Instead, I should move towards a uniform prng such as std::uniform_distribution at some point.

**Comment by bshastry** [comment]
Merging it for now. Looking forward to the template based code improvement you have in mind @ekpyron !

File: test/tools/ossfuzz/protoToAbiV2.cpp [hunk 1]
```diff
@@ -1,10 +1,98 @@
...
+			void                                                               \
+		>                                                                      \
+	>;
+/// Instantiate the using declarations for signed and unsigned integer types.
+BOOST_PP_REPEAT(32, USINGDECL, 1)
+BOOST_PP_REPEAT(32, USINGDECL, 0)
+/// Case implementation that returns an integer value of the specified type.
+/// For signed integers, we divide by two because the range for boost multiprecision
+/// types is double that of Solidity integer types. Example, 8-bit signed boost
+/// number range is [-255, 255] but Solidity `int8` range is [-128, 127]
```
Comment (ekpyron) [inline test/tools/ossfuzz/protoToAbiV2.cpp:32]: By the way that's a **really** weird thing to do from the boost multiprecision side :-).

**Pull request merged**